src/server/publicApi.js

```diff
@@ -10,8 +10,16 @@ import { getTypeString, escapeMarkdownText, isGameModeCornucopia } from '../util
 const runPublicApi = (gameServer) => {
   const app = new Koa();
   const router = new Router();
-  router.get('/players/:id', async (ctx) => {
-    const matchID = ctx.params.id;
+  router.get('/players/:game/:id', async ctx => {
```

Comment:
I would even assume that whatever library is used to make the GET request in the frontend, it will probably natively support to make requests with basicAuth

Reply:
E.g. for superagent you should be able to do something like

request 
  .get('http://local')
  .auth('tobi', 'learnboost')

see https://visionmedia.github.io/superagent/

I would try whether this already sets the Authorization header as wanted.